Range Weapons can compile and be loaded on Clients but not on Servers.

### Issue description

Mod can compile and run on Clients just fine. However, Servers fail to start when trying to load the Mod.
Looking at logs, it mentions something about not being able to execute an entry point on "main" and the class of the range weapon's entity not being loaded in the "SERVER" environment. There might be more things at play but that's just all I see so take my analysis with a gain of salt.

### How to reproduce this issue?

Step 1. Make a Ranged Item
Step 2. Configure it any way you want.
Step 3. Save
Step 4. Run as Server and Client 
Step 5. Observe Logs. The server crashes, but not the Client.

### Operating system

Windows

### Details

MCreator 2022.2.33511
Fabric Generator 1.7-beta-8 (1.19.2)

### Example workspace

[rangedweapontestservercrash.zip](https://github.com/Goldorion/Fabric-Generator-MCreator/files/9523157/rangedweapontestservercrash.zip)


### Logs

```
Executing Gradle task: runClient 
Build info: MCreator 2022.2.33511, fabric-1.19.2, 64-bit, 16332 MB, Windows 10, JVM 17.0.3, JAVA_HOME: C:\Program Files\Pylo\MCreator2022.2\jdk, started on: 2022-09-07-19:37:34 
  
> Configure project : 
Fabric Loom: 0.12.56 
Not publishing sources jar as it was not found. Use java.withSourcesJar() to fix. 
> Task :compileJava UP-TO-DATE 
> Task :processResources UP-TO-DATE 
> Task :classes UP-TO-DATE 
> Task :generateDLIConfig 
> Task :generateLog4jConfig 
> Task :generateRemapClasspath 
> Task :configureLaunch 
> Task :runServer 
19:37.38 [main/INFO] (FabricLoader/GameProvider) Loading Minecraft 1.19.2 with Fabric Loader 0.14.9 
19:37.38 main/INFO (FabricLoader) Loading 35 mods: 
   - fabric-api 0.59.0+1.19.2 
   - fabric-api-base 0.4.10+e62f51a390 
   - fabric-api-lookup-api-v1 1.6.8+9ff28f4090 
   - fabric-biome-api-v1 9.0.16+83a8659290 
   - fabric-command-api-v2 2.1.6+0c17ea9690 
   - fabric-content-registries-v0 3.2.2+83a8659290 
   - fabric-convention-tags-v1 1.1.0+7cd20a1490 
   - fabric-crash-report-info-v1 0.2.4+bd0a0d4a90 
   - fabric-data-generation-api-v1 5.1.7+a680b9b490 
   - fabric-dimensions-v1 2.1.29+a6d2f78590 
   - fabric-entity-events-v1 1.4.17+9ff28f4090 
   - fabric-events-interaction-v0 0.4.27+9ff28f4090 
   - fabric-game-rule-api-v1 1.0.20+1899036190 
   - fabric-gametest-api-v1 1.0.34+e62f51a390 
   - fabric-item-api-v1 1.5.6+35a03c4390 
   - fabric-item-groups-v0 0.3.27+6bee109e90 
   - fabric-lifecycle-events-v1 2.1.1+83a8659290 
   - fabric-loot-api-v2 1.1.2+83a8659290 
   - fabric-message-api-v1 5.0.2+176380a290 
   - fabric-mining-level-api-v1 2.1.13+33fbc73890 
   - fabric-networking-api-v1 1.2.3+5eb68ef290 
   - fabric-object-builder-api-v1 4.0.10+7675279690 
   - fabric-particles-v1 1.0.9+dc39553a90 
   - fabric-registry-sync-v0 0.9.23+23c4cfef90 
   - fabric-rendering-data-attachment-v1 0.3.13+9ff28f4090 
   - fabric-rendering-fluids-v1 3.0.6+9ff28f4090 
   - fabric-resource-conditions-api-v1 2.0.10+e62f51a390 
   - fabric-resource-loader-v0 0.6.0+6bee109e90 
   - fabric-screen-handler-api-v1 1.2.12+9ff28f4090 
   - fabric-transfer-api-v1 2.0.10+83a8659290 
   - fabric-transitive-access-wideners-v1 1.2.0+1e232e1390 
   - fabricloader 0.14.9 
   - java 17 
   - minecraft 1.19.2 
   - test 1.0.0 
19:37.39 main/INFO (FabricLoader/Mixin) SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/C./Users/Micasddsa/.mcreator/gradle/caches/modules-2/files-2.1/net.fabricmc/sponge-mixin/0.11.4+mixin.0.8.5/c1dc27696aa4006e492c2485c9ccbcb26045a971/sponge-mixin-0.11.4+mixin.0.8.5.jar Service=Knot/Fabric Env=SERVER 
19:37.39 [main/INFO] (FabricLoader/Mixin) Loaded Fabric development mappings for mixin remapper! 
19:37.39 [main/INFO] (FabricLoader/Mixin) Compatibility level set to JAVA_16 
19:37.42 main/WARN (FileUtil) Configuration conflict: there is more than one oshi.properties file on the classpath 
> Configure project : 
Fabric Loom: 0.12.56 
19:37.42 main/WARN (FileUtil) Configuration conflict: there is more than one oshi.architecture.properties file on the classpath 
Not publishing sources jar as it was not found. Use java.withSourcesJar() to fix. 
> Task :compileJava UP-TO-DATE 
> Task :processResources UP-TO-DATE 
> Task :classes UP-TO-DATE 
> Task :generateDLIConfig 
> Task :generateLog4jConfig 
> Task :generateRemapClasspath 
> Task :configureLaunch 
19:37.49 [main/INFO] (Minecraft) Building unoptimized datafixer 
> Task :downloadAssets UP-TO-DATE 
> Task :configureClientLaunch 
> Task :runClient 
2022-09-07 19:37:51,944 main ERROR Unable to delete file C:\Users\Micasddsa\MCreatorWorkspaces\test\run\logs\latest.log: java.nio.file.FileSystemException C:\Users\Micasddsa\MCreatorWorkspaces\test\run\logs\latest.log: The process cannot access the file because it is being used by another process 
19:37.52 [main/INFO] (FabricLoader/GameProvider) Loading Minecraft 1.19.2 with Fabric Loader 0.14.9 
19:37.52 [main/INFO] (TestMod) Initializing TestMod 
19:37.52 [main/ERROR] (Minecraft) Failed to start the minecraft server 
java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'test'! 
   at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0([EntrypointUtils.java:51](file:///net.fabricmc.loader.impl.entrypoint.EntrypointUtils:51)) ~[fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions([ExceptionUtil.java:33](file:///net.fabricmc.loader.impl.util.ExceptionUtil:33)) ~[fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0([EntrypointUtils.java:49](file:///net.fabricmc.loader.impl.entrypoint.EntrypointUtils:49)) ~[fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke([EntrypointUtils.java:35](file:///net.fabricmc.loader.impl.entrypoint.EntrypointUtils:35)) ~[fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer([Hooks.java:62](file:///net.fabricmc.loader.impl.game.minecraft.Hooks:62)) ~[fabric-loader-0.14.9.jar:?] 
   at net.minecraft.server.Main.main([Main.java:105](file:///net.minecraft.server.Main:105)) [minecraft-project-@-merged-named.jar:?] 
   at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch([MinecraftGameProvider.java:461](file:///net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider:461)) [fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.launch.knot.Knot.launch([Knot.java:74](file:///net.fabricmc.loader.impl.launch.knot.Knot:74)) [fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.launch.knot.KnotServer.main([KnotServer.java:23](file:///net.fabricmc.loader.impl.launch.knot.KnotServer:23)) [fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.devlaunchinjector.Main.main([Main.java:86](file:///net.fabricmc.devlaunchinjector.Main:86)) [dev-launch-injector-0.2.1+build.8.jar:?] 
Caused by: java.lang.BootstrapMethodError: java.lang.RuntimeException: Cannot load class net.mcreator.test.entity.RangedTestEntity in environment type SERVER 
   at net.mcreator.test.init.TestModEntities.load([TestModEntities.java:23](file:///net.mcreator.test.init.TestModEntities:23)) ~[main/:?] 
   at net.mcreator.test.TestMod.onInitialize([TestMod.java:31](file:///net.mcreator.test.TestMod:31)) ~[main/:?] 
   at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0([EntrypointUtils.java:47](file:///net.fabricmc.loader.impl.entrypoint.EntrypointUtils:47)) ~[fabric-loader-0.14.9.jar:?] 
   ... 7 more 
Caused by: java.lang.RuntimeException: Cannot load class net.mcreator.test.entity.RangedTestEntity in environment type SERVER 
   at net.fabricmc.loader.impl.transformer.FabricTransformer.transform([FabricTransformer.java:59](file:///net.fabricmc.loader.impl.transformer.FabricTransformer:59)) ~[fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPreMixinClassByteArray([KnotClassDelegate.java:454](file:///net.fabricmc.loader.impl.launch.knot.KnotClassDelegate:454)) ~[fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray([KnotClassDelegate.java:407](file:///net.fabricmc.loader.impl.launch.knot.KnotClassDelegate:407)) ~[fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass([KnotClassDelegate.java:323](file:///net.fabricmc.loader.impl.launch.knot.KnotClassDelegate:323)) ~[fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass([KnotClassDelegate.java:218](file:///net.fabricmc.loader.impl.launch.knot.KnotClassDelegate:218)) ~[fabric-loader-0.14.9.jar:?] 
   at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass([KnotClassLoader.java:145](file:///net.fabricmc.loader.impl.launch.knot.KnotClassLoader:145)) ~[fabric-loader-0.14.9.jar:?] 
   at java.lang.ClassLoader.loadClass([ClassLoader.java:520](file:///java.lang.ClassLoader:520)) ~[?:?] 
   at net.mcreator.test.init.TestModEntities.load([TestModEntities.java:23](file:///net.mcreator.test.init.TestModEntities:23)) ~[main/:?] 
   at net.mcreator.test.TestMod.onInitialize([TestMod.java:31](file:///net.mcreator.test.TestMod:31)) ~[main/:?] 
   at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0([EntrypointUtils.java:47](file:///net.fabricmc.loader.impl.entrypoint.EntrypointUtils:47)) ~[fabric-loader-0.14.9.jar:?] 
   ... 7 more 
  
BUILD SUCCESSFUL in 18s 
6 actionable tasks: 4 executed, 2 up-to-date 
Task completed in 18 seconds
```

### Issue tracker rule checks (please read carefully)

- [x] I have checked that my problem <a href='https://github.com/MCreator/MCreator/issues?q=is%3Aissue' target='_blank'>is not already reported</a>
- [X] I have checked that my problem is not covered on <a href='https://mcreator.net/support/knowledgebase' target='_blank'>Knowledge Base</a> or on <a href='https://mcreator.net/wiki' target='_blank'>MCreator's Wiki</a>
- [X] I have checked that my written content does not violate the <a href='https://mcreator.net/wiki/general-publishing-guidelines' target='_blank'>publishing guidelines</a>

--- a/src/fabric-1.19.2/templates/rangeditem/projectile_entity.java.ftl
+++ b/src/fabric-1.19.2/templates/rangeditem/projectile_entity.java.ftl
@@ -27,7 +27,6 @@ package ${package}.entity;
 import net.fabricmc.api.Environment;
 
 <#compress>
-@Environment(EnvType.CLIENT) 
 public class ${name}Entity extends AbstractArrow implements ItemSupplier {
 
 	public ${name}Entity(EntityType<? extends ${name}Entity> type, Level world) {
@@ -42,7 +41,7 @@ public class ${name}Entity extends AbstractArrow implements ItemSupplier {
 		super(type, entity, world);
 	}
 
-	@Override @Environment(EnvType.CLIENT) public ItemStack getItem() {
+	@Override public ItemStack getItem() {
 		<#if !data.bulletItemTexture.isEmpty()>
 			return ${mappedMCItemToItemStackCode(data.bulletItemTexture, 1)};
 		<#else>

